src/features/wallet/root/ActivityItem.tsx

```diff
@@ -1,49 +1,175 @@
-import React from 'react'
+import React, { useMemo, memo, useCallback } from 'react'
 import { TouchableWithoutFeedback } from 'react-native'
-import { formatDistanceToNow } from 'date-fns'
-import { round } from 'lodash'
+import { formatDistanceToNow, fromUnixTime } from 'date-fns'
 import animalHash from 'angry-purple-tiger'
-import { useTheme } from '@shopify/restyle'
+import { useTranslation } from 'react-i18next'
+import {
+  AnyTransaction,
+  RewardsV1,
+  AddGatewayV1,
+  AssertLocationV1,
+  PaymentV2,
+  PaymentV1,
+  PendingTransaction,
+  TransferHotspotV1,
+} from '@helium/http'
 import Box from '../../../components/Box'
 import Text from '../../../components/Text'
 import shortLocale from '../../../utils/formatDistance'
 import Rewards from '../../../assets/images/rewards.svg'
-import { triggerNavHaptic } from '../../../utils/haptic'
-import { Theme } from '../../../theme/theme'
+import SentHnt from '../../../assets/images/sentHNT.svg'
+import HotspotAdded from '../../../assets/images/hotspotAdded.svg'
+import ReceivedHnt from '../../../assets/images/receivedHNT.svg'
+import Location from '../../../assets/images/location.svg'
+import { triggerNotification } from '../../../utils/haptic'
+import { isPayer } from '../../../utils/transactions'
+import { useColors } from '../../../theme/themeHooks'
 
 type Props = {
-  type: string
-  time: number
-  amount?: number
+  item: AnyTransaction | PendingTransaction
   isFirst: boolean
   isLast: boolean
+  address?: string | null
 }
 
-const titles: Record<string, string> = {
-  rewards: 'Mining Rewards',
-  sent: 'Sent HNT',
-  received: 'Received HNT',
-  add: 'Hotspot Added to Blockchain',
-}
+const TxnTypeKeys = [
+  'rewards_v1',
+  'payment_v1',
+  'payment_v2',
+  'add_gateway_v1',
+  'assert_location_v1',
+  'transfer_hotspot_v1',
+] as const
+type TxnType = typeof TxnTypeKeys[number]
 
 const ActivityItem = ({
-  type,
-  time,
-  amount = 0,
+  item,
   isFirst = false,
   isLast = false,
+  address = '',
 }: Props) => {
-  const handlePress = () => {
-    triggerNavHaptic()
-  }
+  const colors = useColors()
+  const { t } = useTranslation()
+
+  const isSending = useMemo(() => isPayer(address, item), [address, item])
```

Comment:
Something is up with this. Mine say Received but they are actually Sent payments

Reply:
Actually it looks like its always false

Reply:
ok, I'm going to try some other addresses and see if I can replicate.

Reply:
Its fixed now